### ISSUE TITLE
request.forms.var returns None if not ascii chars are sent

### ISSUE THREAD

**Issue by boumboum** [bottlepy/bottle#1111]
Hi,
If I query the request.forms parameters where not only ascii characters are submitted it return None, instead of the value (encoded or not encoded)

print(request.forms.myvar) gives None, 
while
print(request.forms.get('myvar')) gives 'óüöúő'

I'm using bottle 0.13-dev

**Comment by defnull** [comment]
Unable to reproduce and this should not be possible.

    request.forms.myvar
    # calls:
    FormsDict.__getattr__('myvar')
    # calls:
    self.getunicode('myvar')
    # calls:
    self._fix(self['myvar'], encoding)

`FormsDict._fix` returns a non-null value in all three branches of its body if `self['myvar']` returned a value. On `UnicodeError` or `KeyError`, `FormsDict.getunicode` returns a default, which is an empty unicode string in this case. `None` can only be returned, if the dict actually contains `None` for the given key, which should never happen for forms parsed from a request, and is ruled out by your example. Either there is something happening in your code that is not part of your example, or I'm missing something.

**Comment by boumboum** [comment]
I know it should not be possible but it is. :-)
Here is an extract from my code:
```
app = Bottle()
@app.post('/send_status')
def send_status():
    comment = request.forms.Comment #gives None which is wrong
    comment = request.forms.get('Comment') #gives óüöúő which is OK
```
Parameters are transmitted as an AJAX call as Multipart form parameters.
If there are only ascii chars and numbers within the params it is working fine

**Comment by defnull** [comment]
What does `request.forms.dict` look like? What happens if you `.decode('utf8')` the value yourself?

**Comment by boumboum** [comment]
request.forms.dict contains well **ALL** the transmitted parameters but nothing is shown if I try to access it via request.forms.Comment.
That's why it took me so long to find the error

**Comment by defnull** [comment]
It's still described here: http://bottlepy.org/docs/dev/tutorial.html#introducing-formsdict (the big yellow box) and in the docstrings of the FormsDict itself.

In short: Item access (e.g. `request.forms['foo']`) will return whatever the WSGI webserver provides, which is a byte string in Python2 and an `latin-1` decoded unicode string in python 3. To get the properly `utf8` decoded value, use attribute access (e.g. `request.forms.foo`) or `request.forms.getunicode('foo')`.

**Comment by defnull** [comment]
I still cannot reproduce or debug this error without knowing whats going on internally, or how exactly the request looked like. Unicode form values are supported and tested. Yours must be different in some way. You could either provide the raw HTTP request (most developer tools in browsers allow you to export these as curl command lines, which would be fine), or you could provide the exact content of `repr(request.forms.dict)`.

**Comment by boumboum** [comment]
Here is the code:
```
from bottle import Bottle, request, response
from cheroot import wsgi

vpeapp = Bottle()

@vpeapp.post('/send_status')
def send_status():
    print('request.forms.get: ', request.forms.get('Comment'))
    print('request.forms.Comment: ', request.forms.Comment)
    print('repr(request.forms.dict): ', repr(request.forms.dict))

if __name__ == '__main__':
    mainapp = Bottle()
    mainapp.mount('/vpe/', vpeapp)
    addr = '0.0.0.0', 9000
    server = wsgi.Server(addr, mainapp)

    try:
        server.safe_start()
    except (KeyboardInterrupt, IOError):
        server.stop()


```
and here are the results:
```
request.forms.get:  óüöúőűáéí
request.forms.Comment:  
repr(request.forms.dict):  {'VPEszam': ['aaxx/245'], 'StatusCode': ['BBB'], 'LoginName': ['abc'], 'Comment': ['óüöúőűáéí'], 'MEMO_id': ['124547'], 'hashcode': ['xyz']}


```
As you can see the the request.form dict is filled well but I cannot retrieve it's content if I'm requesting it in the request.forms.Comment form

**Comment by boumboum** [comment]
And here is the raw data from Insomnia:
```

* Preparing request to http://192.168.100.11:9000/vpe/send_status
* Using libcurl/7.57.0-DEV OpenSSL/1.0.2o zlib/1.2.11 libssh2/1.7.0_DEV
* Enable automatic URL encoding
* Enable SSL validation
* Enable cookie sending with jar of 1 cookie
* Connection 5 seems to be dead!
* Closing connection 5
* Hostname 192.168.100.11 was found in DNS cache
*   Trying 192.168.100.11...
* TCP_NODELAY set
* Connected to 192.168.100.11 (192.168.100.11) port 9000 (#6)
> POST /vpe/send_status HTTP/1.1
> Host: 192.168.100.11:9000
> User-Agent: insomnia/6.2.3
> Content-Type: multipart/form-data; boundary=X-INSOMNIA-BOUNDARY
> Accept: */*
> Content-Length: 522
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="VPEszam"
| aaxx/245
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="StatusCode"
| BBB
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="LoginName"
| abc
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="Comment"
| óüöúőűáéí
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="MEMO_id"
| 124547
| --X-INSOMNIA-BOUNDARY
| Content-Disposition: form-data; name="hashcode"
| xyz
| --X-INSOMNIA-BOUNDARY--
* We are completely uploaded and fine
< HTTP/1.1 200 OK
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< Date: Mon, 10 Dec 2018 18:01:53 GMT
< Server: Cheroot/6.5.2

* Connection #6 to host 192.168.100.11 left intact

```

**Comment by defnull** [comment]
I actually found a bug, but the return value of `request.forms.Comment` should still be an empty string and never `None`

**Comment by sharpaper** [comment]
Has this been resolved? I still seem to have this exact issue. It seems to appear with `<form method="post" enctype="multipart/form-data" ...` The `<input type="text" />` fields returns `request.forms["key"]` as unicode, `request.forms.get("key")` as unicode, `request.forms.key` as None, `request.forms.getunicode("key")` as None.

**Comment by sharpaper** [comment]
For example `<input type="input" name="key" value="jèrry" />`

The content of `request.forms.dict` is

    {
        "key": [
            "j\u00e8rry.svg"
        ]
    }

`request.forms['key']` and `request.forms.get("key")` return `jèrry`, whereas `request.forms.key` and `request.forms.getunicode("key")` return `None`.

**Comment by defnull** [comment]
Fixed in master, not backpacked to 0.12 yet.

**Comment by defnull** [comment]
Let's continue this in #1165
